mail_tracking/models/mail_message.py

```diff
@@ -103,10 +125,66 @@ def _message_read_dict_postprocess(self, messages, message_tree):
         mail_messages = self.browse(mail_message_ids)
         partner_trackings = mail_messages.tracking_status()
         email_cc = mail_messages._get_email_cc()
+        failed_message = self._get_failed_message()
         for message_dict in messages:
             mail_message_id = message_dict.get('id', False)
             if mail_message_id:
-                message_dict['partner_trackings'] = \
-                    partner_trackings[mail_message_id]
-                message_dict['email_cc'] = email_cc[mail_message_id]
+                message_dict.update({
+                    'partner_trackings': partner_trackings[mail_message_id],
+                    'email_cc': email_cc[mail_message_id],
+                    'failed_message': failed_message[mail_message_id],
+                })
         return res
+
+    @api.model
+    def _prepare_dict_failed_message(self, message):
+        failed_trackings = message.mail_tracking_ids.filtered(
+            lambda x: x.state in ['error', 'rejected', 'spam', 'bounced',
+                                  'soft-bounced'])
+        failed_partners = failed_trackings.mapped('partner_id')
+        failed_recipients = [[partner.id, partner.name]
+                             for partner in failed_partners]
+        return {
+            'id': message.id,
+            'date': message.date,
+            'author_id': [message.author_id.id,
+                          message.author_id.display_name],
+            'body': message.body,
+            'subtype_id': self.env.ref(
+                'mail_tracking.mail_message_subtype_failed_retry').id,
```

Comment:
I guess we don't want this to raise an exception if somebody deletes the record by accident, isn't it?

```suggestion
                'mail_tracking.mail_message_subtype_failed_retry', False).id,
```